### PR TITLE
feat: add support for dynamic filename

### LINKED ISSUE #113: Add support for dynamic filename
Currently exporter.setFileName() only supports fixed strings.
It'd be nice to have another method that allows generating filename dynamically, such as:
`exporter.setFileName(()->"exported-"+LocalDateTime.now().format(formatter))`

### PR DESCRIPTION
**Pull request by javier-godoy**
Close #113

## Summary by CodeRabbit

- **New Features**
	- Enhanced file name handling for exported resources, allowing for dynamic file name generation.
	- Introduced a method to append file extensions for exported files.

- **Version Update**
	- Updated the version of the grid-exporter-addon to 2.5.0-SNAPSHOT.

### CODE REVIEW

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/GridExporter.java [hunk 1]
```diff
@@ -307,20 +307,20 @@
...
 
   public GridExporterStreamResource getPdfStreamResource(String template) {
-    return new GridExporterStreamResource(fileName + ".pdf",
+    return new GridExporterStreamResource(getFileName("pdf"),
         makeConcurrentWriter(new PdfStreamResourceWriter<>(this, template)));
   }
 
   public StreamResource getCsvStreamResource() {
-    return new StreamResource(fileName + ".csv", new CsvStreamResourceWriter<>(this));
+    return new StreamResource(getFileName("csv"), new CsvStreamResourceWriter<>(this));
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/GridExporter.java:315]: CSV is not subject to concurrent controls

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/GridExporter.java [hunk 2]
```diff
@@ -114,7 +114,7 @@ public class GridExporter<T> implements Serializable {
 
   String title = "Grid Export";
 
-  String fileName = "export";
+  private SerializableSupplier<String> fileNameSupplier = () -> "export";
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/GridExporter.java:117]: Fixed in setters

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/GridExporter.java [hunk 3]
```diff
@@ -499,7 +503,17 @@
    * @param fileName
    */
   public void setFileName(String fileName) {
-    this.fileName = fileName;
+    fileNameSupplier = () -> fileName;
+  }
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/GridExporter.java:508]: @coderabbitai please check again

**Review by paodb — APPROVED** [review]
LGTM

**Pull request merged**